engine/utils/curl_utils.cc

```diff
@@ -242,8 +242,8 @@ cpp::result<YAML::Node, std::string> ReadRemoteYaml(const std::string& url) {
   }
 }
 
-cpp::result<Json::Value, std::string> SimpleGetJson(const std::string& url,
-                                                    const int timeout) {
+cpp::result<Json::Value, std::string> SimpleGetJson(
```

Comment:
I'm not sure if we adding `recursive` here. The util should just be a simple GET. The user of this API should be in charge to determine if they need to recursive get another API.

And I can't find where in this PR we use this?